### PR TITLE
Fixed incorrect rendering of hidden review fields.

### PR DESCRIPTION
**Pull request by U-DON**
## Description
This fixes an error that was introduced by a refactor on the logic that rendered expand-under review fields:
https://github.com/department-of-veterans-affairs/vets-website/commit/859be14a7d01fa7d361e6b8e811965cc8d6fb1d8#diff-abbfc0d859784c6512383e29e1f95b227785c7975ae3e5a76e4394b6fd13f647L139-R124

Note the mismatch in the first argument of `showField`/`showReviewField` before and after refactoring.

I also refactored the relevant unit tests to use RTL instead of SkinDeep and added unit tests that catch the error.

## Testing done
Confirmed new unit tests failed with the bug and pass with the fix.

## Acceptance criteria
- [ ] Hidden review fields should not be rendered.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms-system/src/js/review/ObjectField.jsx [hunk 1]
```diff
@@ -118,8 +118,14 @@ class ObjectField extends React.Component {
         });
         if (objectFields.length > 1 && visible.length > 0) {
           return objectFields
-            .filter(() =>
-              showReviewField(first, schema, uiSchema, formData, formContext),
+            .filter(propName =>
+              showReviewField(
+                propName,
```
Comment (U-DON) [inline src/platform/forms-system/src/js/review/ObjectField.jsx:123]: This is the main change. The bug was that we were choosing for each field whether to `showReviewField` based only on the top-level field. The fix is to determine whether to `showReviewField` for each field individually.

File: src/platform/forms-system/test/js/review/ObjectField.unit.spec.jsx [hunk 2]
```diff
@@ -368,15 +521,13 @@ describe('Schemaform review: ObjectField', () => {
       />,
     );
 
-    expect(
-      tree.subTree('.form-review-panel-page-header-row').subTree('.edit-btn')
-        .props['aria-label'],
-    ).to.equal('Custom label');
+    expect(tree.getByLabelText('Custom label')).to.exist;
```
Comment (bkjohnson) [inline src/platform/forms-system/test/js/review/ObjectField.unit.spec.jsx:524]: This seems like it is no longer checking `aria-label`. Could we do something like:

```js
expect(tree.querySelector('[aria-label="Custom label"]')).to.exist
```

File: src/platform/forms-system/test/js/review/ObjectField.unit.spec.jsx [hunk 3]
```diff
@@ -406,14 +558,17 @@ describe('Schemaform review: ObjectField', () => {
         onBlur={onBlur}
       />,
     );
-    // expecting a "div.review" instead of a "dl.review"
-    const review = tree.props.children[1];
-    expect(review.type).to.equal('div');
-    expect(review.props.className).to.equal('review');
+
+    expect(tree.getByLabelText('Edit Blah')).to.exist;
```
Comment (bkjohnson) [inline src/platform/forms-system/test/js/review/ObjectField.unit.spec.jsx:562]: Why does this matter?

File: src/platform/forms-system/test/js/review/ObjectField.unit.spec.jsx [hunk 4]
```diff
@@ -443,14 +599,17 @@ describe('Schemaform review: ObjectField', () => {
         onBlur={onBlur}
       />,
     );
-    // expecting a "div.review" instead of a "dl.review"
-    const review = tree.props.children[1];
-    expect(review.type).to.equal('dl');
-    expect(review.props.className).to.equal('review');
+
+    expect(tree.getByLabelText('Edit Blah')).to.exist;
```
Comment (bkjohnson) [inline src/platform/forms-system/test/js/review/ObjectField.unit.spec.jsx:603]: And here too.

File: src/platform/forms-system/test/js/review/ObjectField.unit.spec.jsx [hunk 5]
```diff
@@ -605,15 +777,22 @@ describe('Schemaform review: ObjectField', () => {
...
-    edit.onClick();
+
+    const title = tree.getByRole('heading');
+    expect(title).to.contain.text('Blah');
+    expect(title.className).to.contain('test-title');
+
+    const testProps = document.querySelector('.test-props');
+    const testField = within(testProps).getByRole('textbox');
+    expect(testField).to.exist;
+    expect(testField.value).to.equal('foo');
```
Comment (bkjohnson) [inline src/platform/forms-system/test/js/review/ObjectField.unit.spec.jsx:788]: Would it be better if this somehow included a check involving `testz`?

**Review by bkjohnson — COMMENTED** [review]
A few questions about some tests. :+1: for switching from skin-deep to RTL

File: src/platform/forms-system/test/js/review/ObjectField.unit.spec.jsx (hunk 3 above)
Comment (bkjohnson) [reply src/platform/forms-system/test/js/review/ObjectField.unit.spec.jsx:562]: I don't have a strong opinion - I was just curious

**Review by bkjohnson — APPROVED** [review]
:ok_hand:

**Pull request merged**